### ISSUE TITLE
Usar full text search nas buscas do conteúdo dos arquivos através de outras bases

### ISSUE THREAD

**Issue by anapaulagomes** [DadosAbertosDeFeira/maria-quiteria#124]
Atualmente as bases de dados do [painel](https://github.com/DadosAbertosDeFeira/maria-quiteria/blob/master/datasets/admin.py) tem usado a busca simples, que é mais lenta e cara em termos computacionais. Devemos substituir essa busca pela busca com `search_vector` que usamos para os arquivos.

Bases de dados afetadas:
* CityCouncilMinuteAdmin
* GazetteAdmin
* CityHallBidAdmin

**Comment by andrewsmedina** [comment]
@anapaulagomes temos algum exemplo de busca que está lenta?

**Comment by andrewsmedina** [comment]
Como pode ser visto nesse post: , a forma mais performática em se tratando de full text search com postgres é usando um campo específico para armazenar o search vector e usar um índice GIN ou GiST. 

O projeto já segue essas práticas, como poder ver aqui https://github.com/andrewsmedina/maria-quiteria/blob/main/datasets/models.py#L65 e aqui https://github.com/andrewsmedina/maria-quiteria/blob/main/datasets/models.py#L70

Rodando local as queries de busca tem executado bem rapidamente (é possível verificar isso usando o debug toolbar) e imprimindo a query e o explain da query podemos ver que o índice e o campo com o serch vector estão sendo utilizados:

```
(Pdb) print(queryset.query)
SELECT "datasets_gazette"."id", "datasets_gazette"."created_at", "datasets_gazette"."updated_at", "datasets_gazette"."crawled_at", "datasets_gazette"."crawled_from", "datasets_gazette"."notes", "datasets_gazette"."date", "datasets_gazette"."power", "datasets_gazette"."year_and_edition", "datasets_gazette"."is_legacy" FROM "datasets_gazette" INNER JOIN "datasets_file" ON ("datasets_gazette"."id" = "datasets_file"."object_id" AND ("datasets_file"."content_type_id" = 8)) WHERE "datasets_file"."search_vector" @@ plainto_tsquery(portuguese::regconfig, Secretaria  Municipal da Fazenda) ORDER BY "datasets_gazette"."date" DESC NULLS LAST, "datasets_gazette"."id" DESC
```

```
(Pdb) print(queryset.explain())
Sort  (cost=52.33..52.33 rows=1 width=212)
  Sort Key: datasets_gazette.date DESC NULLS LAST, datasets_gazette.id DESC
  ->  Nested Loop  (cost=40.28..52.32 rows=1 width=212)
        ->  Bitmap Heap Scan on datasets_file  (cost=40.00..44.02 rows=1 width=4)
              Recheck Cond: (search_vector @@ '''secret'' & ''municipal'' & ''fazend'''::tsquery)
              Filter: (content_type_id = 8)
              ->  Bitmap Index Scan on datasets_fi_search__52321c_gin  (cost=0.00..40.00 rows=1 width=0)
                    Index Cond: (search_vector @@ '''secret'' & ''municipal'' & ''fazend'''::tsquery)
        ->  Index Scan using datasets_gazette_pkey on datasets_gazette  (cost=0.28..8.30 rows=1 width=212)
              Index Cond: (id = datasets_file.object_id)
```

Se em produção a pesquisa está com problemas de performance, eu penso em duas hipóteses:
* será que o índice GIN foi criado em produção? Podemos verificar isso com o debug toolbar ou fazendo o explain da query tanto no django como no postgres.
* será que não é outra coisa que está afetando a performance? Isso também pode ser investigado usando o debug toolbar.

@anapaulagomes  essas hipóteses fazem sentido? Como proceder aqui?

**Comment by anapaulagomes** [comment]
O @andrewsmedina descobriu que o índice não está funcionando corretamente em produção (na verdade, não está funcionando). Dado que vamos mudar a infra em breve (🙏🏽), vamos deixar como está por hora e revisitar essa issue quando a migração for feita (#134).

**Comment by geraldo-castro** [comment]
@anapaulagomes Com a mudança de infra feita, cê acha que podemos avaliar a performance dessa busca?

**Comment by anapaulagomes** [comment]
Sim, tudo já foi migrado. 🚀 Pode checar se funciona local e depois na Absam (a galera do canal #infraestrutura do nosso Discord pode dar mais informações e eu posso ajudar com as credenciais tb). @geraldo-castro

**Comment by anapaulagomes** [comment]
Configurei o pghero hoje. Vou deixar alguns dias rodando por lá e posto os resultados aqui.

**Comment by exageraldo** [comment]
Saiu um artigo hoje falando um pouco sobre esse tópico.
Deixando o link aqui pra caso alguem queira entender mais/ganhar mais contexto. 🤗 

[Basic and Full-text Search with Django and Postgres](https://testdriven.io/blog/django-search/)
